GiEnJul/Entities/Connection.cs

```diff
@@ -7,13 +7,41 @@ public Connection()
 
         public Connection(string receiverLocation, string rowKey) : base(receiverLocation, rowKey)
         { }
+        public Connection(Giver giver, Recipient recipient)
+        { 
+            if(giver.PartitionKey != recipient.PartitionKey){
+                throw new System.Exception();
```

Comment:
It's good to think about what kind of exception you are throwing here, so that the application can handle it the proper way.
This might be a good time for creating a custom exception, maybe an InvalidEventException or InvalidLocationException with the message ("Locations/events must match for giver and recipient")?

Reply:
I agree! I would suggest to create an issue for this task and keep it separate from this PR. What do you think?

Reply:
Could do that, but I just really dislike merging this as it is 😨